src/docfx/restore/RestoreMap.cs

```diff
@@ -24,31 +26,30 @@ public string GetGitRestorePath(string url)
         {
             Debug.Assert(!string.IsNullOrEmpty(url));
 
-            var gitRestorePath = s_mappings.GetOrAdd(
-                $"{url}",
-                new Lazy<string>(() =>
-                {
-                    var (remote, branch) = GitUtility.GetGitRemoteInfo(url);
-                    var restoreDir = AppData.GetGitDir(remote);
-
-                    if (!Directory.Exists(restoreDir))
-                    {
-                        throw Errors.NeedRestore(url).ToException();
-                    }
-
-                    return Directory.EnumerateDirectories(restoreDir, "*", SearchOption.TopDirectoryOnly)
-                        .Select(f => PathUtility.NormalizeFolder(f))
-                        .Where(f => f.EndsWith($"{HrefUtility.EscapeUrlSegment(branch)}/"))
-                        .OrderByDescending(f => new DirectoryInfo(f).LastAccessTimeUtc)
-                        .FirstOrDefault();
-                })).Value;
+            var gitRestorePath = s_gitPath.GetOrAdd(url, new Lazy<string>(FindLastModifiedGitRepository)).Value;
 
             if (!Directory.Exists(gitRestorePath))
             {
                 throw Errors.NeedRestore(url).ToException();
             }
 
             return gitRestorePath;
+
+            string FindLastModifiedGitRepository()
+            {
+                var (remote, branch) = GitUtility.GetGitRemoteInfo(url);
+                var restoreDir = AppData.GetGitDir(url);
+
+                if (!Directory.Exists(restoreDir))
+                {
+                    throw Errors.NeedRestore(url).ToException();
+                }
+
+                return Directory.EnumerateDirectories(restoreDir, "*", SearchOption.TopDirectoryOnly)
+                    .Where(f => f.EndsWith(HrefUtility.EscapeUrlSegment(branch)))
+                    .OrderByDescending(f => new DirectoryInfo(f).LastWriteTimeUtc)
+                    .FirstOrDefault();
```

Comment:
lock before using

Reply:
Let's do it in another PR. lock is async, I am tentative to change this method to an async method.